### DISCUSSION TITLE
can't run the code using docker-compose

### DISCUSSION THREAD

**Discussion by gzwine** [shlinkio/discussions#27] [Help wanted] [answered]
#### Shlink web client version

* Version: 3.10.0
* How do you use shlink-web-client: self-hosted

#### Summary

I tried using docker-compose to run the source code. but I get ERROR info from Docker logs like below:
https://stackoverflow.com/questions/66239691/what-does-npm-install-legacy-peer-deps-do-exactly-when-is-it-recommended-wh

It's seems there are some problem with NPM v7+ installs peerDependencies .

#### Current behavior

can't run the web-client

#### Expected behavior

run correctly

#### How to reproduce

1、download the source code 
2、run "docker-compose up -d"
3、check out the docker logs

**Comment by gzwine** [comment]
I fixed this issue by modifying docker composite. yml.

added param "--legacy-peer-deps"

`version: '3'

services:
  shlink_web_client_node:
    container_name: shlink_web_client_node
    image: node:18.12-alpine
    command: /bin/sh -c "cd /home/shlink/www && npm install --force --legacy-peer-deps&& npm run start"
    volumes:
      - ./:/home/shlink/www
    ports:
      - "3000:3000"
      - "56745:56745"
      - "4173:4173"
`

**Comment by acelaya** [comment, marked as answer]
Are you trying to use docker-compose to setup a development environment or a production environment?

If it's the first one, the project includes its own [docker-compose](https://github.com/shlinkio/shlink-web-client/blob/develop/docker-compose.yml) files, which take all this into consideration. Just follow the steps in [contributing](https://github.com/shlinkio/shlink-web-client/blob/f6334c3618daa3c2b3e11fdca30d9a692aa44f5a/CONTRIBUTING.md).

If it's for production environment, don't use the project sources. There's an official `shlink-web-client` docker image, which is production-ready and all dependencies are included https://shlink.io/documentation/shlink-web-client/installation/#docker-image
